tests/Entity/EntitySearchTest.php

```diff
@@ -579,4 +599,40 @@ public function test_search_suggestion_endpoint()
         $this->withHtml($resp)->assertElementCount('a', 0);
         $resp->assertSee('No items available');
     }
+
+    public function test_word_with_punctuation_searchable()
+    {
+        $this->entities->newPage([
+            'name' => 'A Day at the Beach',
+            'html' => '<p>The sun shines brightly over the blue-green-ocean, inviting visitors to relax on the shore.</p><p>The waves crash gently, offering a peaceful rhythm.</p>'
+        ]);
+
+        $this->entities->newPage([
+            'name' => 'Hiking Through the Forest',
+            'html' => '<p>The trail winds through towering trees, with the sound of birds filling the air.</p><p>The blue-green-ocean peeks through the horizon, offering a glimpse of serenity.</p>'
+        ]);
+
+        $this->entities->newPage([
+            'name' => 'Sunset by the Cliffs',
+            'html' => '<p>The sun dips below the horizon, casting warm hues over the blue-green-ocean.</p><p>The scene is tranquil, with a soft breeze brushing the cliffs.</p>'
+        ]);
+
+        $this->entities->newPage([
+            'name' => 'Exploring the City',
+            'html' => '<p>Bustling streets are full of life, with a blend of modern architecture and historical landmarks.</p><p>Cafes and shops line the sidewalks, offering a variety of experiences.</p>'
+        ]);
+
+        $search = $this->asEditor()->get('/search?term=' . urlencode('green ocean'));
+        $output = $search->baseResponse->original['entities'];
+        
+        foreach ($output as $value) {
+            if(!(strpos($value->text,'green-ocean') || strpos($value->name,'green-ocean')))
```

Comment:
@brijm-improwised Test case is just checking happy path for `-`. You should check for `@` , `/` also